exercises/practice/binary-search-tree/BinarySearchTree.cs

```diff
@@ -1,53 +1,10 @@
 using System;
-using System.Collections;
-using System.Collections.Generic;
 
-public class BinarySearchTree : IEnumerable<int>
+public class BinarySearchTree<T> where T : IComparable
 {
-    public BinarySearchTree(int value)
-    {
-    }
+    public int Count => throw new NotImplementedException("You need to implement this function.");
+    public int Depth => throw new NotImplementedException("You need to implement this function.");
```

Comment:
Hi @ErikSchierboom. I agree that `Count` and `Depth` are properties however... The properties are implemented using [lambda expressions](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions) which are used, according to [Microsoft's Documentation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions) used to create anonymous _functions_. 

Both statements are correct. We are asking for the implementation of a property by implementing a function. It gets a bit more obscure if auto properties are involved, but they still have a getter and possibly a setter behind the scenes. 

If both are OK, why am I writing so much instead of clicking "accept suggestion"? I have checked 10 other exercises in the track, and they all, consistently use `"You need to implement this function"`. If you think it would be beneficial to change this to "property" and the below to "method", then probably we should review all the exercises now, or over time for consistency. I'd be happy with either approach, but I wanted to make sure you considered it in the context of the wider track, not only what's in this PR.

Reply:
> if you think it would be beneficial to change this to "property" and the below to "method", then probably we should review all the exercises now

I think so, yes. Function is definitely wrong and should be method. Another PR to update these would be great!